src/Aspire.Hosting/Dashboard/CommandsConfigurationExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal static class CommandsConfigurationExtensions
+{
+    internal static IResourceBuilder<T> WithLifeCycleCommands<T>(this IResourceBuilder<T> builder) where T : IResource
+    {
+        builder.WithCommand(
+            "start",
+            "Start",
+            resourceSnapshot => Task.FromResult(IsStopped(resourceSnapshot.State?.Text)),
+            async (resourceName, services) =>
+            {
+                await services.GetRequiredService<ApplicationExecutor>().StartResourceAsync(resourceName).ConfigureAwait(false);
+            },
+            "<path d=\"M17.22 8.69a1.5 1.5 0 0 1 0 2.62l-10 5.5A1.5 1.5 0 0 1 5 15.5v-11A1.5 1.5 0 0 1 7.22 3.2l10 5.5Zm-.48 1.75a.5.5 0 0 0 0-.88l-10-5.5A.5.5 0 0 0 6 4.5v11c0 .38.4.62.74.44l10-5.5Z\"/>",
```

Comment:
I think we need to allow for custom icons but what safety measures do we need here.

Reply:
Mind you ... there is way more you could do if you have code that is executing here. No need to use a HTML injection attack.